setup/mail-postfix.sh

```diff
@@ -232,11 +232,32 @@ tools/editconf.py /etc/postfix/main.cf \
 # As a matter of fact RFC is not strict about retry timer so postfix and
 # other MTA have their own intervals. To fix the problem of receiving
 # e-mails really latter, delay of greylisting has been set to
-# 180 seconds (default is 300 seconds).
+# 180 seconds (default is 300 seconds). We will move the postgrey database
+# under $STORAGE_ROOT. This prevents a "warming up" that would have occured
+# previously with a migrated or reinstalled OS.  We will specify this new path
+# with the --dbdir=... option. Arguments within POSTGREY_OPTS can not have spaces,
+# including dbdir. This is due to the way the init script sources the
+# /etc/default/postgrey file. --dbdir=... either needs to be a path without spaces
+# (luckily $STORAGE_ROOT does not currently work with spaces), or it needs to be a
+# symlink without spaces that can point to a folder with spaces).  We'll just assume
+# $STORAGE_ROOT won't have spaces to simplify things.
 tools/editconf.py /etc/default/postgrey \
-	POSTGREY_OPTS=\"'--inet=127.0.0.1:10023 --delay=180'\"
+	POSTGREY_OPTS=\""--inet=127.0.0.1:10023 --delay=180 --dbdir=$STORAGE_ROOT/mail/postgrey/db"\"
 
 
+# If the $STORAGE_ROOT/mail/postgrey is empty, copy the postgrey database over from the old location
+if [ ! -d $STORAGE_ROOT/mail/postgrey/db ]; then
+	# Stop the service
+	service postgrey stop
+	# Ensure the new base path for postgrey exists
+	mkdir -p $STORAGE_ROOT/mail/postgrey
+	# Move over database
+	mv /var/lib/postgrey $STORAGE_ROOT/mail/postgrey/db
```

Comment:
Getting back to this after a long time, I think this line will have an error in initial installs since /var/lib/postgrey won't exist?

Reply:
I'll need to verify.  We do stop the service, so my assumption is that after postgrey is installed, it is started.  If so, it should create files in /var/lib/postgrey that would be mv'ed without a problem on a new install or existing.

Reply:
I can confirm that after we apt_install postgrey, the service is started and creates initial files in /var/lib/postgrey.  I've added additional creation of mkdir -p $STORAGE_ROOT/mail/postgrey/db and  || true the mv.